### PR TITLE
Updated the cloud block service support. With new changes for AWS.

### LINKED ISSUE #389: [Multicloud UI] Enhancement of the Multi-Cloud Block Support for AWS Create/List/Get/Update/Delete
**Issue/Feature Description:**
In the Faroe release a basic multicloud volume service was introduced. 
For the user to use the multucloud Block service the following will have to be provided:

-  Register Backend 

-  List Backend

 - Create Volume for AWS Backend (With more types and parameters. Tracked in #390 )

 -  List all Volume

 -  List Volume for AWS Backend

-  Get Volume

 - Update Volume for AWS Backend (Being tracked in #399 )

 - Volume FileShare

**Why this issue to fixed / feature is needed(give scenarios or use cases):**

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
This PR adds new functionality and enhances the current Cloud block service features and support for AWS backend.
User can now create different types of AWS Volumes. The different types of volumes are preconfigured with default settings and can be modified as per user requirements. There are min and max checks on size and iops as per AWS.

This PR does the following:
- [x] Create Volume for AWS Backend
- [x] Create Volume for AWS Backend based on different types

- [x] List all Volume

- [x] List Volume for AWS Backend

- [x] Get Volume

- [x] Delete Volume

**Which issue(s) this PR fixes**:

Fixes #389  #390 

**Test Report Added?**:
/kind TESTED

**Test Report**:

**List All Volumes**
list-volume--all-types

**Expand Volume details**
list-volume--expand

**Create Volume page**
create-volume-1

**Create Volume Different Volume type options**
create-volume-types-gp

**Create Volume Types**
create-volume-types

**Create Volume - Provisioned IOPS**
create-volume-types-io

**Create Volume - Creating status in expand details**
create-volume-details-creating

**Create Encrypted Volume**
create-volume-encrypted

**Create Volume - Max value validation for size and iops**
create-volume-types-io-max-validation

**Create Volume - Min value validation for size and iops**
create-volume-types-io-min-validation

**Create Volume Success - Creating status**
create-volume-success-creating

**Delete Volumes**
delete-volumes

**Delete Volumes Success**
delete-volumes-success

**AWS Cloud backend Volume listing**
Volumes-EC2-Management-Console

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by kumarashit** [comment]
IOPS may not be mandatory for all types of volume, is it verified?

File: src/app/shared/utils/consts.ts [hunk 1]
```diff
@@ -60,8 +61,9 @@ export const Consts = {
         'ibm-cos': "IBM COS",
         'yig': "YIG Ceph",
         'alibaba-oss' : "Alibaba Object Storage",
-        'aws-file' : 'AWS File Service',
-        'azure-file' : 'Azure File Service'
+        'aws-file' : 'AWS File Storage',
+        'azure-file' : 'Azure File Storage',
+        'aws-block' : 'AWS Block Storage'
```
Comment (kumarashit) [inline src/app/shared/utils/consts.ts:66]: gcp-file name not defined?

Comment (anvithks) [reply src/app/shared/utils/consts.ts:66]: That is part of the GCP file share PR. This PR only is only for the Block storage service.

**Comment by anvithks** [comment]
> IOPS may not be mandatory for all types of volume, is it verified?

Yes. Where not mandatory the validation is removed.

File: src/app/shared/utils/consts.ts (hunk 1 above)
Comment (kumarashit) [reply src/app/shared/utils/consts.ts:66]: But in CLOUD_TYPE you have defined the gcp-file. So half part is here

**Comment by kumarashit** [comment]
> > IOPS may not be mandatory for all types of volume, is it verified?
> 
> Yes. Where not mandatory the validation is removed.

Can you please, give here for which all types of volume, IOPS are taken? From code i see only io1 takes InputText for IOPS. Maybe not clear to me.
Also, what is the max value of IOPS 64k?

File: src/app/shared/utils/consts.ts (hunk 1 above)
Comment (anvithks) [reply src/app/shared/utils/consts.ts:66]: :facepalm: let me check and remove that.
i have pushed the changes as part of #414 .

**Comment by anvithks** [comment]
> > > IOPS may not be mandatory for all types of volume, is it verified?
> > 
> > 
> > Yes. Where not mandatory the validation is removed.
> 
> Can you please, give here for which all types of volume, IOPS are taken? From code i see only io1 takes InputText for IOPS. Maybe not clear to me.
> Also, what is the max value of IOPS 64k?

@kumarashit 
As per the AWS Volume creation page IOPS can be provided for General Purpose (gp2) and Provisioned IOPS (io1 and io2).
IOPS is mandatory only for Provisioned IOPS. For General Purpose it is not so removed the field.

As per AWS the min value for IOPS is 100 and max is 64000 as shown in the images below. 
For General purpose there is a calculation in AWS as follows:
> _(Baseline of 3 IOPS per GiB with a minimum of 100 IOPS, burstable to 3000 IOPS)_

Currently this calculation is not applied in our code. 
I will update the PR.

**General Purpose**
image
**Provisioned IOPS**
image

For the rest, COLD HDD (sc1), Throughput Optimized HDD (st1) and Magentic (standard) the IOPS field is not applicable so it is not displayed and not mandatory.

**Cold HDD**
image

**Throughput Optimised**
image

**Magnetic**
image

**Comment by kumarashit** [comment]
Thanks @anvithks for the details.
Please check this too
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-volume-types.html

**Review by kumarashit — APPROVED** [review]
LGTM

**Comment by anvithks** [comment]
> Thanks @anvithks for the details.
> Please check this too
> https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-volume-types.html

@kumarashit Tracking this in #415

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**